coreax/coreset.py

```diff
@@ -131,26 +147,20 @@ class Coresubset(Coreset[_Data], Generic[_Data]):
     :param pre_coreset_data: The dataset :math:`X` used to construct the coreset.
     """
 
-    # Incompatibility between Pylint and eqx.field. Pyright handles this correctly.
-    # pylint: disable=no-member
+    def __init__(self, nodes: Data, pre_coreset_data: _Data):
+        """Handle typing of ``nodes`` being a `Data` instance."""
+        # I do not think there is a way here to satisfy Pyright in the current
+        # implementation of this class as the ``nodes`` here *must* be a `Data`
+        # instance and cannot be generic on `_Data` as is the case in the parent class.
+        super().__init__(nodes, pre_coreset_data)
```

Comment:
This does get rid of the Pyright complaint, but it implies that the type of `pre_coreset_data` should be `Data` and not `SupervisedData`, which it could be.

Reply:
Not the end of the world due to the inheritance but doesn't feel as clear as I would like

Reply:
I can make the change and raise an issue? What do you think

Reply:
>  it implies that the type of pre_coreset_data should be Data and not SupervisedData, which it could be.

SupervisedData is a subclass of Data, so any function that accepts Data should also accept SupervisedData.

Reply:
That is very helpful, thanks. I agree now!